ArticleTemplates/assets/js/modules/messenger.js

```diff
@@ -0,0 +1,198 @@
+/* global Promise */
```

Comment:
We need to support offline reading so we'd have to bundle the polyfill with the assets served from the device. This is something I'm going to look at this afternoon as this PR is dependant on the bundled polyfill being available.